### PR TITLE
make it so dynamic and static can live together in NiChrome

### PR DESCRIPTION
**Pull request by rminnich**
Add a new directory, /bbin, and put busybox builds there. 
Have init search  globs, not paths, so that we can pick up, e.g., NiChrome commands. 

Have installcommand use the same glob for finding command source.

For now, since we hit a bug in the Go toolchain (22819), re enable printing
commands from Go build. This bug is supposed to be fixed in 1.10.

    This has been tested with NiChrome and seems to work ok.

    add pack to the tools

    This is disappointing, but it seems we need a 5th binary, pack,
     see 22189

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: cmds/installcommand/installcommand.go [hunk 1]
```diff
@@ -171,16 +172,29 @@ func main() {
 	// and the glob will find it. If we get to the point that the glob
 	// no longer works we can go with filepath.Walk (which glob uses anyway)
 	// but for now this works.
-	src := filepath.Join("/src", util.CmdsPath, form.cmdName)
-	if _, err := os.Stat(src); err != nil {
-		l, err := filepath.Glob(filepath.Join("/src/*/*/", form.cmdName))
-		if err != nil || len(l) == 0 {
-			log.Fatalf("Can't find source code for %v (err %v)", form.cmdName, err)
+	var src string
+	for _, p := range util.CmdsGlob {
```
Comment (hugelgupf) [inline cmds/installcommand/installcommand.go]: Why not just filepath.Walk /src and match on any directory with the cmdName? Seems a little more robust than this glob.

File: pkg/uroot/source.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func buildToolchain(opts BuildOpts) error {
 	}
 
 	toolDir := filepath.Join(opts.TempDir, fmt.Sprintf("go/pkg/tool/%v_%v", opts.Env.GOOS, opts.Env.GOARCH))
-	for _, pkg := range []string{"compile", "link", "asm"} {
+	for _, pkg := range []string{"pack", "compile", "link", "asm"} {
```
Comment (hugelgupf) [inline pkg/uroot/source.go]: pack necessary? the tool doesn't build with -x -a by default, i think. if so, let's have that in a different commit.

File: cmds/installcommand/installcommand.go (hunk 1 above)
Comment (rminnich) [reply cmds/installcommand/installcommand.go]: glob uses Walk, so from my POV, this is a lot less code that gets the same effect. I really prefer it.

File: pkg/uroot/source.go (hunk 2 above)
Comment (rminnich) [reply pkg/uroot/source.go]: This is a workaround that goes away in 1.10, we hope, but yeah I can move it to a separate commit. 

That said, I think when we demo this thing, it's nice to have the output again for a while. It really test the point across that we're compiling. 

For now, I will remove pack, set the verbose back to false, and that will clean this up.

File: cmds/installcommand/installcommand.go (hunk 1 above)
Comment (hugelgupf) [reply cmds/installcommand/installcommand.go]: not the same effect -- if i were to add a command from say github.com/u-root/NiChrome/cmds/foo, that doesn't match the glob anymore, does it?

Comment (rminnich) [reply cmds/installcommand/installcommand.go]: btw, I'll take your word for it, we can fix this, but I'd like to get this PR in as is and have the discussion later. I think the glob is still more useful but maybe I'm totally wrong :-)

**Pull request merged**